buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/BuildPlugin.groovy

```diff
@@ -200,6 +230,15 @@ class BuildPlugin implements Plugin<Project>  {
         project.sourceCompatibility = '1.8'
         project.targetCompatibility = '1.8'
 
+        // TODO: Remove all root project distribution logic. It should exist in a separate dist project.
+        if (project != project.rootProject) {
+            SourceSet mainSourceSet = project.sourceSets.main
+            FileCollection javaSourceDirs = mainSourceSet.java.sourceDirectories
+            javaSourceDirs.each { File srcDir ->
```

Comment:
is this loop correct ?  I would have expected to see something like:
```
project.getArtifacts().add('sourceElements', mainSourceSet.java.sourceDirectories)
```
(no loop)

Reply:
I don't believe that will work since `sourceDirectories` is a `FileCollection`. The `ArtifactHandler.add()` doesn't accept collections. You need to add the directories individually.

That said, because of this, we are open to configuration ordering issues here. So any additional source directories need to be added _before_ this code runs.

Reply:
Yeah, I wasn't able to find anything on FileCollections that allows you to listen for changes like other domain object collections. I think anything outside of the default java directory needs to be added to the source set explicitly, and thus should be getting added to the sourceElements configuration explicitly as well.